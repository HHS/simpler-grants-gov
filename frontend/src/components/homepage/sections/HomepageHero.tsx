import { ExternalRoutes } from "src/constants/routes";

import { useTranslations } from "next-intl";
import Link from "next/link";
import { GridContainer } from "@trussworks/react-uswds";

import { USWDSIcon } from "src/components/USWDSIcon";

const Hero = () => {
  const t = useTranslations("Homepage");

  return (
    <div
      data-testid="homepage-hero"
      className="hero bg-primary-darkest text-white overflow-hidden"
    >
      <GridContainer className="hero--container padding-y-1 tablet:padding-y-3 tablet-lg:padding-y-10 desktop-lg:padding-top-15 desktop-lg:padding-bottom-10 position-relative">
        <div className="hero--text position-relative z-100">
          <h1 className="text-ls-neg-2 tablet:font-sans-2xl desktop-lg:font-sans-3xl desktop-lg:margin-top-2 text-balance">
            <span>{t("pageTitle")}</span>
          </h1>
          <p className="usa-intro line-height-sans-3 font-sans-md tablet:font-sans-lg text-balance">
            {t("pageDescription")}
          </p>
        </div>
        <span className="usa-dark-background bg-transparent">
          <Link
            className="usa-button usa-button--small usa-button--unstyled usa-button--inverse desktop:position-absolute top-3 right-0 margin-y-1 desktop:margin-y-3 desktop:margin-x-4"
            href={ExternalRoutes.GITHUB_REPO}
            target="_blank"
          >
            <USWDSIcon
              name="github"
              className="usa-icon usa-icon--size-3"
              aria-label="Github"
            />
            {t("github_link")}
          </Link>
        </span>
      </GridContainer>
    </div>
  );
};

export default Hero;
