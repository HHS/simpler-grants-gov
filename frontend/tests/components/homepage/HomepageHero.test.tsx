import { render, screen } from "tests/react-utils";

import HomepageHero from "src/components/homepage/sections/HomepageHero";

describe("Homepage Hero Content", () => {
  it("matches snapshot", () => {
    const { container } = render(<HomepageHero />);
    expect(container).toMatchSnapshot();
  });
  it("Renders without errors", () => {
    render(<HomepageHero />);
    const component = screen.getByTestId("homepage-hero");

    expect(component).toBeInTheDocument();
  });
});
