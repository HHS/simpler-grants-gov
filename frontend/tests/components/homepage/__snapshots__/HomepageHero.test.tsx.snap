// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Hero Content matches snapshot 1`] = `
<div>
  <div
    class="hero bg-primary-darkest text-white overflow-hidden"
    data-testid="homepage-hero"
  >
    <div
      class="grid-container hero--container padding-y-1 tablet:padding-y-3 tablet-lg:padding-y-10 desktop-lg:padding-top-15 desktop-lg:padding-bottom-10 position-relative"
      data-testid="gridContainer"
    >
      <div
        class="hero--text position-relative z-100"
      >
        <h1
          class="text-ls-neg-2 tablet:font-sans-2xl desktop-lg:font-sans-3xl desktop-lg:margin-top-2 text-balance"
        >
          <span>
            Let's build a simpler Grants.gov together
          </span>
        </h1>
        <p
          class="usa-intro line-height-sans-3 font-sans-md tablet:font-sans-lg text-balance"
        >
          Simpler.Grants.gov is our testing ground for the next generation of Grants.gov. With your help, we're setting a new standard for transparency and usability in government services. 
        </p>
      </div>
      <span
        class="usa-dark-background bg-transparent"
      >
        <a
          class="usa-button usa-button--small usa-button--unstyled usa-button--inverse desktop:position-absolute top-3 right-0 margin-y-1 desktop:margin-y-3 desktop:margin-x-4"
          href="https://github.com/HHS/simpler-grants-gov"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            class="usa-icon usa-icon usa-icon--size-3"
            role="img"
          >
            <use
              href="/img.jpg#github"
            />
          </svg>
          Follow on GitHub
        </a>
      </span>
    </div>
  </div>
</div>
`;
